Automattic/wp-calypso
# Menus: fix randomly closing item edit view

Observing `sites` in menus/main.jsx forces a re-render whenever `sites` changes.
When this happens, the edit-tracking `lock` object is created anew, causing any currently
open item edit box to close.

This change removes the use of the lock lib entirely, using state to track
the currently edited item.

**To Reproduce**
1) Go to https://wordpress.com/menus and select a site
2) Click on the edit icon of an existing menu item
<img width="38" alt="screen shot 2015-12-03 at 12 44 43" src="https://cloud.githubusercontent.com/assets/7767559/11560941/bbadee3e-99bb-11e5-8b10-71209f2b6bc6.png">
3) The edit view will open for the item
4) Wait a while, and the edit view will close on its own

**To Test**
With this PR, all menus behaviour should be the same, except for the edit view closing on its own
